projects/igniteui-angular/src/lib/carousel/carousel.component.ts

```diff
@@ -505,13 +511,8 @@ export class IgxCarouselComponent implements OnDestroy, AfterContentInit {
         if (this.previousSlide && this.previousSlide.previous) {
             this.previousSlide.previous = false;
         }
-        if (this.leaveAnimationPlayer) {
+        if (this.leaveAnimationPlayer || this.enterAnimationPlayer) {
```

Comment:
I would put this logic in separate method, e.g. `animationStarted`.
Now you have `finishAnimations` method which does not finish animations anymore but returns if the any animation is started.
Go like this leave `finishAnimation` as it was. Add `animationStarted` method and call it where you need to check if any animation was started.